### PR TITLE
Add option for single vs multi OncoKB icons

### PR DESCRIPTION
**Pull request by onursumer**
Related to cBioPortal/cbioportal#8710.

We also need to a backend PR to add a server side config property.

## Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

### CODE REVIEW

**Comment by phillis2** [comment]
image

should say: OncoKB: a precision oncology knowledge base that contains...

**Comment by phillis2** [comment]
it was not clear to me that the target icon was clickable or useful for changing the display

File: packages/react-mutation-mapper/src/component/oncokb/OncoKB.tsx [hunk 1]
```diff
@@ -149,6 +154,86 @@ export default class OncoKB extends React.Component<IOncoKbProps, {}> {
...
+                            type={dataType}
+                            tooltipOverlay={this.tooltipContent(dataType)}
+                            indicator={this.props.indicator}
+                            availableDataTypes={this.props.availableDataTypes}
+                        />
+                    ))}
+                </>
+            );
+        } else {
+            // TODO This doesn't always work, in some cases it adds unnecessary empty icons, we need a better solution.
```
Comment (onursumer) [inline packages/react-mutation-mapper/src/component/oncokb/OncoKB.tsx:197]: this issue (cBioPortal/cbioportal#9141) will be addressed in a separate PR.

File: packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    Px2: '#1F78B4',
+    Px3: '#984EA3',
+};
+
+function getLevelColor(
+    highestLevel?: string,
+    defaultColor: string = '#CCCCCC'
+) {
+    const level = normalizeLevel(highestLevel || null);
+    return LEVEL_COLOR_MAP[level || ''] || defaultColor;
```
[the remark this answers is not in the document]
Comment (onursumer) [reply packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx:45]: We get `Type 'null' cannot be used as an index type.` error since level can be null. We can do the `|| ''` on the line above maybe.

File: packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+) {
+    return (
+        level &&
+        availableDataTypes !== undefined &&
+        availableDataTypes.includes(type)
+    );
+}
+
+function getOncogenicShape(indicator?: IndicatorQueryResp) {
+    const oncogenicity = oncogenicityIconClassNames(
```
[the remark this answers is not in the document]
Comment (onursumer) [reply packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx:61]: Actually, this function is designed to return class names for the icon elements. For example see https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/shared/components/mutationTable/column/annotation/AnnotationHeader.tsx#L196

I am just reusing it here to get the oncogenciity level. I think ideally we should have a separate function which just returns the level.

File: src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx [hunk 4]
```diff
@@ -374,4 +380,15 @@ export default class CopyNumberTableWrapper extends React.Component<
             </div>
         );
     }
+
+    @action.bound
+    private handleOncoKbToggle(mergeIcons: boolean) {
+        this.mergeOncoKbIcons = mergeIcons;
+        updateOncoKbIconStyle({ mergeIcons });
+
+        calculateOncoKbContentWidthOnNextFrame(
```
[the remark this answers is not in the document]
Comment (onursumer) [reply src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx:389]: this is the combined width of the OncoKB related icons within the column.

File: src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx [hunk 5]
```diff
@@ -68,34 +75,27 @@ type ICopyNumberTableWrapperProps = {
...
-                    Number(
-                        document
-                            .getElementById('copy-number-annotation')
-                            ?.getElementsByClassName('oncokb-content')[0]
-                            ?.clientWidth
-                    ) || 22;
-                clearInterval(this.oncokbInterval);
-            }
-        }, 500);
+        this.oncokbInterval = calculateOncoKbContentWidthWithInterval(
```
[the remark this answers is not in the document]
Comment (onursumer) [reply src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx:93]: This is an existing code which I just refactored to reduce code duplication. The reason is to adjust the annotation column header alignment after the column cells finish rendering. Using an interval is probably not the best solution. Maybe we can fix this in a follow up PR?

**Review by leexgh — COMMENTED** [review]
@onursumer Oncokb toggle seems can change column selection. I believe this is not what we want, right?
Dec-22-2021 12-38-40

File: packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (onursumer) [reply packages/react-mutation-mapper/src/component/oncokb/CompactAnnotationIcon.tsx]: replaced this function with a new one called `normalizeOncogenicity`

File: src/pages/patientView/structuralVariant/PatientViewStructuralVariantTable.tsx [hunk 6]
```diff
@@ -458,4 +461,15 @@ export default class PatientViewStructuralVariantTable extends React.Component<
     public render() {
         return this.tableUI.component;
     }
+
+    @action.bound
+    private handleOncoKbToggle(mergeIcons: boolean) {
+        this.mergeOncoKbIcons = mergeIcons;
+        updateOncoKbIconStyle({ mergeIcons });
+
+        calculateOncoKbContentWidthOnNextFrame(
```
[the remark this answers is not in the document]
Comment (onursumer) [reply src/pages/patientView/structuralVariant/PatientViewStructuralVariantTable.tsx:473]: added a comment but haven't changed the function name. the function itself does not actually set the width, it just calculates the width and passes it to the callback function. `set` might be misleading in the function name.

**Comment by onursumer** [comment]
> @onursumer Oncokb toggle seems can change column selection. I believe this is not what we want, right? Dec-22-2021 12-38-40

Good catch! Feels like the problem is with the `reset columns` event.  I'll look into it.

**Comment by inodb** [comment]
@tmazor could you take a look at this one?

**Comment by tmazor** [comment]
Looks great to me @inodb @onursumer @alisman !
